substratevm/src/com.oracle.svm.jni/src/com/oracle/svm/jni/hosted/JNINativeCallWrapperSubstitutionProcessor.java

```diff
@@ -37,17 +41,30 @@
  */
 class JNINativeCallWrapperSubstitutionProcessor extends SubstitutionProcessor {
     private final Map<ResolvedJavaMethod, JNINativeCallWrapperMethod> callWrappers = new ConcurrentHashMap<>();
+    private final DuringSetupAccessImpl access;
+
+    JNINativeCallWrapperSubstitutionProcessor(DuringSetupAccessImpl access) {
+        super();
+        this.access = access;
+    }
 
     @Override
     public ResolvedJavaMethod lookup(ResolvedJavaMethod method) {
         assert method.isNative() : "Must have been registered as a native substitution processor";
+        String jniJavaCallWrappersInternalName = MetaUtil.toInternalName(JNIJavaCallTrampolines.class.getTypeName());
```

Comment:
Since you already have a `DuringSetupAccessImpl` through which you can get a `MetaAccessProvider`, please lookup `JNIJavaCallTrampolines.class` as an `AnalysisType` once in the constructor, store it in a field and do a simple comparison by reference on `method.getDeclaringClass()`.

Reply:
Done